File: app/src/main/java/com/android/bookingcucimotor/DetailActivity.java
package com.android.bookingcucimotor;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.android.bookingcucimotor.databinding.ActivityDetailBinding;

public class DetailActivity extends AppCompatActivity {

    private ActivityDetailBinding binding;
    private Booking dataBooking;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityDetailBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.detail_activity), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle(getString(R.string.detail_page));
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
        }

        dataBooking = getIntent().getParcelableExtra("key_booking");

        if (dataBooking != null) {
            if (dataBooking.getImg() != -1) {
                binding.ivDetailImg.setImageResource(dataBooking.getImg());
            } else {
                binding.ivDetailImg.setImageResource(R.drawable.baseline_image_24);
            }

            binding.tvDetailName.setText(dataBooking.getName());
            binding.tvDetailPrice.setText(dataBooking.getPrice());
            binding.tvDetailDescription.setText(dataBooking.getDescription());
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: app/src/main/java/com/android/bookingcucimotor/DetailActivity.kt
package com.android.bookingcucimotor

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.android.bookingcucimotor.databinding.ActivityDetailBinding

@Suppress("DEPRECATION")
class DetailActivity : AppCompatActivity() {

    private lateinit var binding: ActivityDetailBinding
    private lateinit var dataBooking: Booking

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.detail_activity)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        supportActionBar?.apply {
            title = getString(R.string.detail_page)
            setDisplayHomeAsUpEnabled(true)
            setDisplayShowHomeEnabled(true)
        }

        dataBooking = intent.getParcelableExtra("key_booking")!!

        dataBooking.let {
            if (it.img != -1) {
                binding.ivDetailImg.setImageResource(it.img)
            } else {
                binding.ivDetailImg.setImageResource(R.drawable.baseline_image_24)
            }

            binding.tvDetailName.text = it.name
            binding.tvDetailPrice.text = it.price
            binding.tvDetailDescription.text = it.description
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}